torch/ao/quantization/pt2e/representation/rewrite.py

```diff
@@ -140,19 +178,29 @@ def _reference_dequantize_per_tensor_int8(x_i8, scale, zero_point, quant_min, qu
     # TODO: debug the implementation later when torchdynamo time out issue is resolved
     return ((x_i8.to(torch.float32) - zero_point) * scale).to(dtype=torch.float32)
 
+# boolean flag to indicate if we want to replace literals to be placeholder nodes for pattern and replacement
+_REPLACE_LITERAL = True
+_DONT_REPLACE_LITERAL = False
+
 _EXAMPLE_INPUTS_PATTERN_AND_REPLACEMENTS = [
-    (_QUANTIZED_ADD_EXAMPLE_INPUTS, _qdq_quantized_add_relu, _reference_quantized_add_relu),
-    (_QUANTIZED_ADD_EXAMPLE_INPUTS, _qdq_quantized_add, _reference_quantized_add),
-    (_QUANTIZE_PER_TENSOR_INT8_EXAMPLE_INPUTS, _quantize_per_tensor_int8, _reference_quantize_per_tensor_int8),
-    (_DEQUANTIZE_PER_TENSOR_INT8_EXAMPLE_INPUTS, _dequantize_per_tensor_int8, _reference_dequantize_per_tensor_int8),
+    (_QUANTIZED_ADD_EXAMPLE_INPUTS, _qdq_quantized_add_relu, _reference_quantized_add_relu, _DONT_REPLACE_LITERAL),
+    (_QUANTIZED_ADD_EXAMPLE_INPUTS, _qdq_quantized_add, _reference_quantized_add, _DONT_REPLACE_LITERAL),
+    (_QUANTIZED_MAX_POOL2D_EXAMPLE_INPUTS, _qdq_quantized_max_pool2d, _reference_quantized_max_pool2d, _REPLACE_LITERAL),
+    (_QUANTIZE_PER_TENSOR_INT8_EXAMPLE_INPUTS, _quantize_per_tensor_int8, _reference_quantize_per_tensor_int8, _DONT_REPLACE_LITERAL),
+    (_DEQUANTIZE_PER_TENSOR_INT8_EXAMPLE_INPUTS, _dequantize_per_tensor_int8, _reference_dequantize_per_tensor_int8, _REPLACE_LITERAL),
 ]
 
 def reference_representation_rewrite(model: GraphModule) -> GraphModule:
     remove_tensor_overload_for_qdq_ops(model)
-    for example_inputs, pattern, replacement in _EXAMPLE_INPUTS_PATTERN_AND_REPLACEMENTS:
+    for example_inputs, pattern, replacement, replace_literals in _EXAMPLE_INPUTS_PATTERN_AND_REPLACEMENTS:
         pattern = get_aten_graph_module(pattern, example_inputs)  # type: ignore[arg-type, assignment]
         remove_tensor_overload_for_qdq_ops(pattern)  # type: ignore[arg-type]
         replacement = get_aten_graph_module(replacement, example_inputs)  # type: ignore[arg-type, assignment]
         remove_tensor_overload_for_qdq_ops(replacement)  # type: ignore[arg-type]
+        if replace_literals:
+            pattern = replace_literals_with_placeholders(pattern)
```

Comment:
you said, this is already not working, right?

Reply:
this is working, just a bit hacky